### PR TITLE
test against Sequence not list (#7673 retargeted at 4.5.x)

### LINKED ISSUE #7672: Ruamel Yaml CommentedSeq Subclasses MutableSequence not List
## Current Behavior
`conda` checks if something is a list, but since it is loaded by `ruamel_yaml` it is a `CommentedSeq`, which now no longer subclasses from `list` but from `MutableSequence`.

```python
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/conda/cli/main_config.py", line 30, in execute
    execute_config(args, parser)
  File "/opt/conda/lib/python3.6/site-packages/conda/cli/main_config.py", line 263, in execute_config
    raise CouldntParseError("key %r should be a list, not %s." % (key, bad))
conda.exceptions.CouldntParseError: key 'channels' should be a list, not CommentedSeq.
```
I think this is the relevant line in master. https://github.com/conda/conda/blob/master/conda/cli/main_config.py#L286

[Commit in `ruamel.yaml` changing inheritance](https://bitbucket.org/ruamel/yaml/commits/9e4de94ee6a88ed86ab5fdc6d49a23cfc6ab5782)

### Steps to Reproduce
Run a build with `ruamel_yaml>0.15.54`.

## Expected Behavior
Not error with `ruamel_yaml>0.15.54`

## Environment Information
Working on it.
<details open><summary><code>`conda info`</code></summary><p>

```

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```

```
</p></details>

### PR DESCRIPTION
**Pull request by mbargull**
Closes #7672

### CODE REVIEW

**Comment by mbargull** [comment]
I restarted the `ci/circleci: 2.1 conda-build ` tests two times and they (somewhat) consistently fail. Those are unrelated of course. But by the way: can we remove/phase out the `conda-build 2.1` tests? Since conda-forge and Bioconda use `conda-build >=3` for some time by now, I don't know who else is still using `2.1` :man_shrugging:

**Comment by kalefranz** [comment]
https://github.com/conda/conda/pull/7673#issuecomment-414322463

File: conda/cli/main_config.py [hunk 1]
```diff
@@ -257,7 +257,8 @@ def execute_config(args, parser):
             if key not in sequence_parameters:
                 from ..exceptions import CondaValueError
                 raise CondaValueError("Key '%s' is not a known sequence parameter." % key)
-            if not isinstance(rc_config.get(key, []), list):
+            if not (isinstance(rc_config.get(key, []), Sequence) and not
+                    isinstance(rc_config.get(key, []), string_types)):
                 from ..exceptions import CouldntParseError
                 bad = rc_config[key].__class__.__name__
                 raise CouldntParseError("key %r should be a list, not %s." % (key, bad))
```
[the remark this answers is not in the document]
Comment (kalefranz) [reply conda/cli/main_config.py]: Our configuration also uses the terms primitive, sequence, and map. However, especially for the sequence type, I think I prefer the user-facing term be list here. Hopefully people who actually know the difference will excuse the blurring of distinction, and those who have no idea what "sequence" means won't end up filing a support issue.

**Comment by kalefranz** [comment]
@CJ-Wright Are you intending to sign the [CLA](https://conda.io/docs/contributing.html#conda-contributor-license-agreement)? I don't have a copy on record for you yet. If not, we'll probably have to figure something else out.

**Comment by CJ-Wright** [comment]
@kalefranz do you have an ETA for this (I don't want to be pushy, but I do want to know so we can plan our mitigation efforts).

**Comment by mbargull** [comment]
Nice, thanks @CJ-Wright, @kalefranz. It probably makes sense to test this out with `conda-forge/label/ruamel_yaml>=0.15.55`, if someone has time to do so.

**Comment by kalefranz** [comment]
Hey @CJ-Wright, I'll try to do it today, since you asked nicely 😜. For `defaults` stuff, we try to put new builds of all of conda's dependencies in `conda-canary` for a week or so before we actually release them to `defaults`, to hopefully help us catch stuff like this.

Speaking of, I guess we should ensure we have a regression test for this.

**Comment by kalefranz** [comment]
Oh, I guess it's failing on `conda config --add channels defaults`. We definitely have that test case already covered.  So the tests would have caught this if we released the dependency to `conda-canary` first.

**Comment by kalefranz** [comment]
As @mbargull mentions, would be nice to have confirmation that this fixes the problem before making the release.  Maybe I could ask @jjhelmus super nicely to update `ruamel_yaml` in `AnacondaRecipes` and upload it to `conda-canary`? 🙏

File: conda/cli/main_config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbargull) [reply conda/cli/main_config.py]: That's why @CJ-Wright added
`isinstance(..., Sequence) and not isinstance(..., string_types)`

[the remark this answers is not in the document]
Comment (kalefranz) [reply conda/cli/main_config.py]: You're right.

**Comment by mbargull** [comment]
> It probably makes sense to test this out with `conda-forge/label/broken::ruamel_yaml>=0.15.55`

Did that. Works perfectly fine with `conda-forge/label/broken::ruamel_yaml=0.15.57`.
Tested with
```
$ conda config --add channel conda-forge
$ conda config --add channel conda-forge
```
which fails for `4.5.10` but works on current head of `4.5.x`!

**Comment by jjhelmus** [comment]
> Maybe I could ask @jjhelmus super nicely to update ruamel_yaml in AnacondaRecipes and upload it to conda-canary?

ruamel_yaml 0.15.60 packages are now available on the conda-canary channel

**Pull request merged**